dataclients/kubernetes/clusterclient.go

```diff
@@ -168,17 +169,22 @@ func newClusterClient(o Options, apiURL, ingCls, rgCls string, quit <-chan struc
 
 	if o.KubernetesInCluster {
 		c.tokenProvider = secrets.NewSecretPaths(time.Minute)
-		err := c.tokenProvider.Add(serviceAccountDir + serviceAccountTokenKey)
-		if err != nil {
-			log.Errorf("Failed to Add secret %s: %v", serviceAccountDir+serviceAccountTokenKey, err)
-			return nil, err
+		c.tokenFile = serviceAccountDir + serviceAccountTokenKey
+	} else if o.TokenFile != "" {
+		c.tokenProvider = secrets.NewSecretPaths(time.Minute)
+		c.tokenFile = o.TokenFile
+	}
+
+	if c.tokenProvider != nil {
+		if err := c.tokenProvider.Add(c.tokenFile); err != nil {
+			return nil, fmt.Errorf("failed to add secret %s: %v", c.tokenFile, err)
```

Comment:
Is it okay to change letter case here? Users may grep by "Failed to Add" and lost this log message.

Reply:
I just make is all consistent. Return error instead of logging and make it a proper sentence - skipper will not start if error happens. Also we do not guarantee log format, at least here (maybe we should be more careful about proxy logs).

Reply:
Let's have an issue and discuss pros and cons and what would be the exact rule we want to apply.
Normally errors should be found somewhere in the log, but I guess here we have some info/debug log and an error with different case-sensitive characters. I am not sure why these have to be in-sync, but maybe there's some great idea here.